pkg/render/manager_route_config.go

```diff
@@ -0,0 +1,416 @@
+		}
+
+		if route.Spec.CABundle != nil {
+			path, err := builder.mountConfigMapReference(route.Spec.CABundle.Name, route.Spec.CABundle.Key)
+			if err != nil {
+				return nil, err
+			}
+
+			r.CABundlePath = path
+		}
+
+		// Require that either both MTLSCert and MTLSKey are set or neither are.
+		if (route.Spec.MTLSCert != nil && route.Spec.MTLSKey == nil) || (route.Spec.MTLSKey != nil && route.Spec.MTLSCert == nil) {
+			return nil, fmt.Errorf("must set both MTLSCert and MTLSKey, or neither for TLS terminated route %s", route.Name)
+		}
+
+		if route.Spec.MTLSCert != nil {
+			path, err := builder.mountSecretReference(route.Spec.MTLSCert.Name, route.Spec.MTLSCert.Key)
+			if err != nil {
+				return nil, err
+			}
+			r.ClientCertPath = path
+
+			// At this point, if MTLSCert is set then MTLSKey will be set otherwise this if statement wouldn't be executed
+			// or an error would have already been returned.
+			path, err = builder.mountSecretReference(route.Spec.MTLSKey.Name, route.Spec.MTLSKey.Key)
+			if err != nil {
+				return nil, err
+			}
+			r.ClientKeyPath = path
+		}
+	}
+
+	for _, route := range builder.tlsPassThroughRoutes {
+		r := &tlsPassThroughRoute{
+			Destination: route.Spec.Destination,
+			ServerName:  route.Spec.SNIMatch.ServerName,
+		}
+		tunnelTLSPassThroughRoutes = append(tunnelTLSPassThroughRoutes, r)
+	}
+
+	routesData := map[string]string{}
+
+	if len(uiTLSTerminatedRoutes) > 0 {
+		jsonBytes, err := json.MarshalIndent(uiTLSTerminatedRoutes, "", "\t")
+		if err != nil {
+			return nil, err
+		}
+
+		routesData[uiTLSTerminatedRoutesKey] = string(jsonBytes)
+	}
+
+	if len(tunnelTLSTerminatedRoutes) > 0 {
+		jsonBytes, err := json.MarshalIndent(tunnelTLSTerminatedRoutes, "", "\t")
+		if err != nil {
+			return nil, err
+		}
+
+		routesData[upstreamTunnelTLSTerminatedRoutesKey] = string(jsonBytes)
+	}
+
+	if len(tunnelTLSPassThroughRoutes) > 0 {
+		jsonBytes, err := json.MarshalIndent(tunnelTLSPassThroughRoutes, "", "\t")
+		if err != nil {
+			return nil, err
+		}
+
+		routesData[upstreamTunnelTLSPassThroughRoutesKey] = string(jsonBytes)
+	}
+
+	// Add the config map so mountConfigMap can add the annotation for the data.
+	builder.AddConfigMap(&corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: voltronRoutesConfigMapName,
+		},
+		Data: routesData,
+	})
+
+	if _, err := builder.mountConfigMapReference(voltronRoutesConfigMapName, ""); err != nil {
+		return nil, err
+	}
+
+	return &VoltronRouteConfig{
+		routesData:   routesData,
+		volumeMounts: builder.volumeMounts,
+		volumes:      builder.volumes,
+		annotation:   rmeta.AnnotationHash(builder.hashableContent),
+	}, nil
+}
+
+func (builder *voltronRouteConfigBuilder) mountConfigMapReference(name, key string) (string, error) {
+	defaultMode := int32(420)
+
+	configMap := builder.configMaps[name]
+	if configMap == nil {
+		return "", fmt.Errorf("the contents for ConfigMap '%s' wasn't provided, and is needed to generate annotations", name)
```

Comment:
"the contents _weren't_ provided" or just "ConfigMap X wasn't provided"

similarly

"and are needed to generate"

Reply:
updated.